### ISSUE TITLE
Cannot Deserialize JSON with `@JsonProperty`-annotated Record fields

### ISSUE THREAD

**Issue by juulhobert** [FasterXML/jackson-jr#176]
I'm encountering an issue when using Jackson-jr with annotation support to handle record classes that map to JSON with snake_case field names. While serialization works as expected, deserialization of the same JSON back into the record fails.

To illustrate the problem, I created a minimal reproducer: [jackson-jr-annotations-reproducer](https://github.com/juulhobert/jackson-jr-annoations-reproducer).

**Steps to Reproduce:**
Use a record class where fields are annotated with @JsonProperty to specify the snake_case equivalent of the field name. Serialize an instance of the record class to JSON using Jackson-jr (with annotation support). Attempt to deserialize the resulting JSON back into the record class.

Expected Behavior:
The JSON should deserialize correctly into the record class, matching fields based on the @JsonProperty annotations.

Actual Behavior:
Deserialization fails with an exception
![image](https://github.com/user-attachments/assets/378c4f7e-9d8c-4b2d-9d61-34f572bc6887)

Environment:
Jackson-jr version: 2.18.1
Java version: 21

Reproducer Link:
https://github.com/juulhobert/jackson-jr-annoations-reproducer

**Comment by cowtowncoder** [comment]
Right, Record-support is pretty limited and probably won't yet work with annotation support. Thank you for reporting this.

**Comment by cowtowncoder** [comment]
Fixed for 2.21.0 and 3.0.1.
